test/e2e/node/pods.go

```diff
@@ -282,6 +282,7 @@ var _ = SIGDescribe("Pods Extended", func() {
 								events = append(events, event)
 								if event.Type == watch.Error {
 									framework.Logf("watch error seen for %s: %#v", pod.Name, event.Object)
+									break
```

Comment:
Interestingly enough, this should never happen, because if we get a watch.Error the watch should be closed.  So it is unexpected to go into a forever loop here unless something is broken with watches and clients.

Second, the error event should not get added to the events array and should actually fail the test.  So this should be roughly:

```
if event.Type == watch.Error {
  ...
  break
}
events = append(events, event)
if event.Type == watch.Deleted {
```

We need to make sure to print out the event object so we can actually see what the failure was.  The verification loop below ignores it and returns success, which is bad.

Reply:
yeah... I'm not sure about this change. I added that logging just to get details if it happened, not because I wanted us to bail out

Reply:
Actually, i need to review this a bit more.  It looks like the verifyFn should be returning this as an error, which would fail the test.  But some errors have to be ignored.